modules/irsa/variables.tf

```diff
@@ -65,3 +59,18 @@ variable "irsa_iam_permissions_boundary" {
   default     = ""
   description = "IAM Policy ARN for IRSA IAM role permissions boundary"
 }
+
+variable "addon_context" {
+  type = object({
+    aws_caller_identity_account_id = string
+    aws_caller_identity_arn        = string
+    aws_eks_cluster_endpoint       = string
+    aws_partition_partition        = string
```

Comment:
While I understand why its named this way should we just call this `aws_partition_name`?

Reply:
agree

Reply:
I think we should suffix the output variable name which is more meaningful here. What @Zvikan did is correct for `aws_partition_partition` (e.g., <resource_name>_<output_variable_name>) . I understand your concern around repeating the partition twice here. 

We can use `aws_partition_id` instead which gives the same output as `aws_partition_partition`

Reply:
Understand the confusion here, but as Vara said we want to follow a naming convention mentioned, this one is a little bit tricky because of how the repeated name and the schema [here](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/meta/partition_data_source.go#L15-L29).
Looking at the [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition) instead of the actual schema code, the `id` is indeed the same so I'll rename it to `aws_partition_id`